harness/go/harness/httpproxy.go

```diff
@@ -82,21 +82,14 @@ func (h *HTTPConnectProxyServer) handler(w http.ResponseWriter, r *http.Request)
 	if err != nil {
 		panic("hijack failed")
 	}
-	res := &http.Response{
-		StatusCode: http.StatusOK,
-		Proto:      "HTTP/1.1",
-		ProtoMajor: 1,
-		ProtoMinor: 1,
-		Header:     make(http.Header),
-	}
 
 	targetConn, err := net.Dial("tcp", r.URL.Host)
 	if err != nil {
 		http.Error(w, fmt.Sprintf("Upstream conn failed: %v", err), http.StatusBadGateway)
 		return
 	}
 
-	if err := res.Write(clientConn); err != nil {
+	if _, err := clientConn.Write([]byte("HTTP/1.1 200 Connection established\r\n\r\n")); err != nil {
```

Comment:
Hrmm, can you clarify what the problem was with the response written previously? I am just curious.

Reply:
Two things:
1. Headers were fragmented, each line of header being sent in its own packet. See in the image below packets 13, 15 and 16.
2. It was sending a `Content-Length: 0` header (packet number 15 in the snapshot), which [is forbidden by the HTTP CONNECT spec](https://www.rfc-editor.org/rfc/rfc9110#section-9.3.6) (though the spec also mandates that clients should ignore that header if present).

For some unknown reason, node's HTTP2 implementation (or maybe grpc-js?) was randomly failing the HTTP2 connection soon after establishment (see the `GOAWAY` at packet 37). I don't know exactly why the two factors contributed occurrence of the `GOAWAY` issue (or either both were contributing to it), but I observed that writing the 200 string in a single operation was significantly reducing the frequency of failure. The status string come from [mitmproxy](https://github.com/mitmproxy/mitmproxy), which I used to compare behavior while investigating this issue.

<img width="741" alt="Screenshot 2024-05-08 at 13 20 20" src="https://github.com/temporalio/features/assets/6412169/a5feac35-73e0-45a1-ba18-6cab1e861d96">

Reply:
Interesting, thanks!